oshmem/runtime/oshmem_shmem_finalize.c

```diff
@@ -70,8 +71,10 @@ int oshmem_shmem_finalize(void)
         /* Should be called first because ompi_mpi_finalize makes orte and opal finalization */
         ret = _shmem_finalize();
 
-        if ((OSHMEM_SUCCESS == ret) && ompi_mpi_initialized
-                && !ompi_mpi_finalized) {
+        int32_t state = ompi_mpi_state;
```

Comment:
@jsquyres I see that atomics are used to change ompi_mpi_state, don't we need to use them when accessing it as well?
Or consideration is made that this will always be the same thread?

Reply:
See the comment starting at https://github.com/open-mpi/ompi/pull/5217/files#diff-b109613962ca7154bf7d060024f15fe2R196.

Reply:
:+1: